otelconf/config.go

```diff
@@ -58,6 +62,50 @@ var noopSDK = SDK{
 	shutdown:       func(context.Context) error { return nil },
 }
 
+var sdk *SDK
+
+// init checks the local environment and uses the file set in the variable
+// `OTEL_EXPERIMENTAL_CONFIG_FILE` to configure the SDK automatically.
+func init() {
+	// look for the env variable
+	filename, ok := os.LookupEnv("OTEL_EXPERIMENTAL_CONFIG_FILE")
+	if !ok {
+		return
+	}
+	b, err := os.ReadFile(filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Parse a configuration file into an OpenTelemetryConfiguration model.
+	c, err := ParseYAML(b)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create SDK components with the parsed configuration.
+	s, err := NewSDK(WithOpenTelemetryConfiguration(*c))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Set the global providers.
+	otel.SetTracerProvider(s.TracerProvider())
+	otel.SetMeterProvider(s.MeterProvider())
+	global.SetLoggerProvider(s.LoggerProvider())
+	sdk = &s
+}
+
+// Shutdown calls the shutdown function of the global SDK instantiated if
+func Shutdown(ctx context.Context) {
+	if sdk == nil {
+		return
+	}
+	if err := sdk.Shutdown(ctx); err != nil {
+		log.Fatal(err)
```

Comment:
We do not want to crash a user application just because this fails. Should an error be logged instead of fatal?

Reply:
code has been removed